### PR TITLE
Fixed GMX ruleset

### LINKED ISSUE #2630: rule gmx.net wrong forward
As fas as I see is the first test of the rule gmx.ch where I end up often. But I'm registered at gmx.net. So that I have to enter the https-adress at the first time manualy.

Please change it that after entering gmx.net I come to https://gmx.net.

Thanks

### PR DESCRIPTION
**Pull request by rugk**
Fixes #2630

### CODE REVIEW

File: src/chrome/content/rules/GMX.xml [hunk 1]
```diff
@@ -1,54 +1,41 @@
 <!--
-
```
Comment (jsha) [inline src/chrome/content/rules/GMX.xml]: Please don't remove the whitespace.

**Comment by rugk** [comment]
Added whitespace again.

File: src/chrome/content/rules/GMX.xml [hunk 2]
```diff
@@ -2,7 +2,6 @@
 
 	For other United Internet coverage, see United-Internet.xml.
 
-
```
Comment (jsha) [inline src/chrome/content/rules/GMX.xml]: It looks like you left some whitespace deletions in.

File: src/chrome/content/rules/GMX.xml [hunk 3]
```diff
@@ -134,12 +123,10 @@
 		<test url="http://www.gmx.ch/" />
 		<test url="http://www.gmx.co.uk/" />
 		<test url="http://www.gmx.fr/" />
+		<test url="http://www.gmx.de/" />
```
Comment (jsha) [inline src/chrome/content/rules/GMX.xml]: These test URLs are unnecessary: They are implicitly generated by the `<target host ...>` tags.

**Comment by jsha** [comment]
I don't see any changes to this ruleset that affect its funcitonality. Am I missing something?

File: src/chrome/content/rules/GMX.xml (hunk 2 above)
Comment (rugk) [inline src/chrome/content/rules/GMX.xml]: These whitespaces are not really needed.
I've just removed the double whitespace line. :wink:

File: src/chrome/content/rules/GMX.xml (hunk 3 above)
Comment (rugk) [inline src/chrome/content/rules/GMX.xml]: Deleted.

File: src/chrome/content/rules/GMX.xml [hunk 4]
```diff
@@ -10,45 +9,37 @@
 
 	Nonfunctional domains:
 		- (www.)gmx.ca		(→ www.gmx.com)
-
```
Comment (jsha) [inline src/chrome/content/rules/GMX.xml]: What about here? It's much easier to review your changes when you change only the parts that need fixing, and don't include additional formatting changes. If you feel like the rule needs a formatting overhaul, I suggest doing that in a separate pull request.

**Comment by rugk** [comment]
So I've added the whitespace again and removed a redirect rule, which was unnecessary.

File: src/chrome/content/rules/GMX.xml (hunk 4 above)
Comment (jsha) [inline src/chrome/content/rules/GMX.xml]: You've added back the missing lines, but now they have spaces at the beginning of them, which still generates unnecessary diff lines. I would recommend starting from a fresh branch, making only the necessary changes, and starting a fresh PR.

**Comment by rugk** [comment]
SO what about this? AFAIK another gmx.net PR was merged, which also changes the things I've changed here, so should this be closed?

**Comment by J0WI** [comment]
You need to merge your branch with current master.
